### ISSUE TITLE
Additional potential formats

### ISSUE THREAD

**Issue by bilalshaikh42** [biosimulators/Biosimulators_utils#89]
Here are some additional file ending encountered
- [x] CSS 
- [x] PSD - photoshop files
- [x] php - php source code
- [x] phps
- [ ] cgi - binary application/webservers
- [x] pl - perl source
- [x] odt - open document format
- [x] Java - java source
- [x] ico - icon 
- [x] graphml - graphml
- [x] class - java 
- [ ] classpath - java? 
- [ ] prefs - must be some sort of settings file
- [ ] kss - some sort of template? 
- [ ] dig - windows help?? 
- [x] mat  - Matlab
- [x] owl - Ontology
- [x] db - sql db? 
- [x] jar - java binary
- [x] sh - text
- [x] mov - video 
- [x] swf - shockwave flash
- [x] xsl - xml style sheet
- [x] as - action script
- [x] fla - Adobe flash
- [x] sql

**Comment by jonrkarr** [comment]
Is this now a complete list?

**Comment by bilalshaikh42** [comment]
This includes all of the files I have encountered via downloading PMR
workspaces. There are a few others that I have decided to filter out of the
workspaces instead. These are

- [ ] .gitignore files
- [ ] IntelliJ workspace files
- [ ] pyc - compiled python files
- [ ] DS_Store - some sort of mac desktop cache
- [ ] hgsub - mercurial submodules (this one is a bit problematic as it may
contain info on how to get some of the missing files)
- [ ] session - Not sure what these do. These are usually of the form
.cellml.session which makes me think they are temp save files from opencor
desktop

From the logs here are some more  that I have not looked up yet

```
Unknown file format: smd
Unknown file format: smd
Unknown file format: template
Unknown file format: example-php
Unknown file format: fla
Unknown file format: sql
Unknown file format: sql
Unknown file format: smd
Unknown file format: smd
Unknown file format: smd
Unknown file format: smd
Unknown file format: smd
Unknown file format: obj1
Unknown file format: res
Unknown file format: bigQuery5
Unknown file format: phps
Unknown file format: obj4
Unknown file format: obj1testArray
Unknown file format: bigQuery
Unknown file format: smd
Unknown file format: query
Unknown file format: obj3
Unknown file format: obj1lazyValue
Unknown file format: manifest
Unknown file format: TODO
Unknown file format: smd
Unknown file format: smd
```

**Comment by bilalshaikh42** [comment]
This includes all of the files I have encountered via downloading PMR workspaces. There are a few others that I have decided to filter out of the workspaces instead. These are:

- [ ] .gitignore files
- [ ] IntelliJ workspace files
- [ ] pyc - compiled python files
- [ ] DS_Store - some sort of mac desktop cache
- [ ] hgsub - mercurial submodules (this one is a bit problematic as it may contain info on how to get some of the missing files)
- [ ] session - Not sure what these do. These are usually of the form .cellml.session which makes me think they are temp save files from opencor desktop

From the logs here are some more  that I have not looked up yet

```
Unknown file format: smd
Unknown file format: smd
Unknown file format: template
Unknown file format: example-php
Unknown file format: fla
Unknown file format: sql
Unknown file format: sql
Unknown file format: smd
Unknown file format: smd
Unknown file format: smd
Unknown file format: smd
Unknown file format: smd
Unknown file format: obj1
Unknown file format: res
Unknown file format: bigQuery5
Unknown file format: phps
Unknown file format: obj4
Unknown file format: obj1testArray
Unknown file format: bigQuery
Unknown file format: smd
Unknown file format: query
Unknown file format: obj3
Unknown file format: obj1lazyValue
Unknown file format: manifest
Unknown file format: TODO
Unknown file format: smd
Unknown file format: smd
```

**Comment by jonrkarr** [comment]
Regarding the extensions you haven't looked up yet, from the BioSimulators-utils perspective, I think we should ignore these non-standard extensions and use the URI `http://purl.org/NET/mediatypes/application/octet-stream` for files with such extensions.
- bigQuery: presumbably related to Google's BigQuery
- bigQuery5: presumbably related to Google's BigQuery
- example-php: this looks like a non-standard extension for PHP source code
- manifest: maybe a text or XML file
- obj1
- obj1lazyValue
- obj1testArray
- obj3
- obj4
- query
- res
- smd: not sure what this is
- template
- TODO: likely a text file

**Comment by jonrkarr** [comment]
I can take care of the list above. I'll likely skip a few that don't seem like standard extensions.

**Comment by jonrkarr** [comment]
I added the formats [above](https://github.com/biosimulators/Biosimulators_utils/issues/89#issue-1091644192) with these exceptions
- The non-standard extensions noted [above](https://github.com/biosimulators/Biosimulators_utils/issues/89#issuecomment-1003444288)
- (Java) `.classpath`: this is typically an environment variable. I don't think there's a canonical media type
- `.db` is also an extension used for SQL scripts. This seems to be used somewhat interchangeably with `.sql`
- `.cgi` is typically an extension for C or Perl. Its not really a distinct media type.
- I couldn't find canonical media types for these
  - IntelliJ workspace files
  - Mercurial submodules
- Not sure what these are
  - `.dig` 
  - `.kss` 
  - `.prefs`: these are probably specific to a particular program
- I agree these seem good to ignore from archives
  -`.gitignore`
  - `.session`
  - `DS_STORE`
  - `pyc`

**Comment by jonrkarr** [comment]
Added in f6797fd9b7b02898e5f6236a110f2a27091e9355. Will be released with 0.1.154.

**Comment by jonrkarr** [comment]
Also added to biosimuations ontology library.
